irohad/ametsuchi/impl/block_query_base.hpp

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_BLOCK_QUERY_BASE_HPP
+#define IROHA_BLOCK_QUERY_BASE_HPP
+
+#include <soci/soci.h>
+
+#include "ametsuchi/block_query.hpp"
+#include "ametsuchi/block_storage.hpp"
+#include "logger/logger_fwd.hpp"
+
+namespace iroha::ametsuchi {
+
+  /**
+   * Class which implements BlockQuery with a Postgres backend.
+   */
+  class BlockQueryBase : public BlockQuery {
+   public:
+    BlockQueryBase(BlockStorage &block_storage, logger::LoggerPtr log);
+
+    BlockResult getBlock(
+        shared_model::interface::types::HeightType height) override;
+
+    shared_model::interface::types::HeightType getTopBlockHeight() override;
+
+    void reloadBlockstore() override;
+
+    std::optional<TxCacheStatusType> checkTxPresence(
+        const shared_model::crypto::Hash &hash) override;
+
+    // res > 0 => Committed
```

Comment:
It would be much better to use `enum struct`

Reply:
No, this code must be compatible with postgres impl. ```res``` is the result of SQL query that way.